lib/parser/message.h

```diff
@@ -67,14 +68,17 @@ constexpr MessageFixedText operator""_err_en_US(
 }
 
 // The construction of a MessageFormattedText uses a MessageFixedText
-// as a vsnprintf()-like formatting string that is applied to the
-// following arguments.  Additional formatting codes are accepted in
-// messages:
-//   %B - the corresponding argument is a CharBlock
-//   %S - the corresponding argument is a std::string
+// as a vsnprintf() formatting string that is applied to the
+// following arguments.  CharBlock and std::string values are also
+// accepted; they are automatically converted to char pointers
+// that are suitable for '%s' formatting.
 class MessageFormattedText {
 public:
-  MessageFormattedText(MessageFixedText, ...);
+  template<typename... A>
+  MessageFormattedText(const MessageFixedText &text, A &&... x)
+    : isFatal_{text.isFatal()} {
+    Format(text, Convert(std::move(x))...);
```

Comment:
Maybe the `std::move(x)` should be a  `std::forward<A>(x)` due to the perfect forwarding context. That said, `Convert()` are not "stealing" any storage from there arguments from what I see, so there should not be any issue. But it also means that maybe taking rvalue args does not bring much here in the end compare to the simpler `const A&`.